src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -280,6 +296,28 @@ private JObject UpdateDependenciesProperty(JObject projectJsonRoot, JObject upda
             return projectJsonRoot;
         }
 
+        private JObject FilterRuntimes(JObject projectRoot, HashSet<string> excludedRIDs)
+        {
+            var runtimes = projectRoot["runtimes"];
+            if (runtimes != null)
+            {
+                var toRemove = new List<JToken>();
+                foreach (JProperty runtime in runtimes)
+                {
+                    if (excludedRIDs.Contains(runtime.Name))
+                        toRemove.Add(runtime);
+                }
+
+                foreach (var token in toRemove)
+                {
+                    Log.LogMessage("Removing RID '{0}' from the list of applicable runtimes.", ((JProperty)token).Name);
+                    token.Remove();
```

Comment:
Well if I knew the type of runtimes then I might be able to give you more definitive guidance (I hate var :(), but a common pattern is to use a for loop in reverse order to do these type of removals.